### DISCUSSION TITLE
Looking Forward - Project Updates & The End of Milestone 1

### DISCUSSION THREAD

**Discussion by arek-kwapis** [Abstrat-Technologies/rustedanvil#29] [Announcements]
Hello folks! I'm here and alive with an update to how the project is going 👋

It's no secret we're reaching the end of the first development phase, Milestone 1. We have managed to document what we believe are all the issues with the project and how to fix said issues. It's been a tedious journey of redoing everything from scratch, and I am glad it's over. However, just because the first phase was done relatively quickly does not mean that the rest of development will go at this pace - phase 2, Milestone 2, will take **considerably longer**. This is because we will develop all the tools necessary to automate the download and fixing of everything we have put down on paper. 

Here in phase 2, we have decided on a few things you all should know about

- The project will be written in Javascript, on the NodeJS framework
  - This is because NodeJS allows us to do everything required, and is malleable enough to any issues that may arise or extra requirements. It's also the language and framework I'm most comfortable and knowledgeable in, so why jump into something new when the old works just as well.
- We are not reworking Forge. We are using it as-is with in-place fixes
  - This isn't the project for this, and I won't pretend to understand what the hell Forge does half the time with how badly these early versions are written.
- There is no time limit
  - We are not in a rush to finish this project. This is a passion project, with a set list of goals. Don't ask for how long something may take, otherwise you'll fall into the "Ask stupid questions get stupid responses" pit. The eventual heat death of the universe is the only "time limit" we have
- The team is small
  - Currently, we only have 1 developer and 1 graphic designer focused on this project. Don't expect the unrealistic.
- The end result will be binaries, not source
  - Due to how NodeJS works and how this project will be developed, executable binaries will be the end result. This deserves a mention due to the fact some people are (quite rightly) apprehensive about running unknown exe's on their PC. The source code will remain public as stated in the license, but the end result that's usable will unfortunately be a executable
- We're only working on Windows (for now)
  - Due to the limited team size and other factors (such as cost of acquiring machines), we are solely focused on development for Windows at the time being. If you'd like it for a different system of your choice, you can just fork it and do it.

Well, that's about it for the summary of what we're doing and working on just now! Keep an eye out, as we'll be updating the repo with new commits and information regularly as development proceeds. Hopefully I can do a "devblog" style post like this one every few weeks with updates on what's going on and whatnot, we'll have to see about that.

Thanks!

P.S: Development will be ultra slow for the next week or so, here's your warning, don't moan at me for it. The team took a short break out of town a few days ago and while we had fun, it seems we brought back with us sickness. Our graphic designer has been out cold ever since we got back and I have personally been dying of the worst kind of flu - man flu. I'm sure development will speed up again once I stop trying to sneeze my lungs out and gobbling on hundreds of flu meds 🤮

**Comment by Unloaded123456** [comment]
The source code is unexecutable in almost all of projects, source needs to be compiled to get an executable program. 

The executable is not readable unless the person has the ability to read raw data, executable needs to be decompiled {and usually deobfuscated} to be readable.

Edit: Seeing the end of Milestone 1 is good.
